Backend/DataAccessLayer/Repositories/SessionRepository.cs

```diff
@@ -0,0 +1,66 @@
+﻿using DataAccessLayer.Models;
+using DataAccessLayer.Database;
+using System;
+using System.Data.Entity;
+using System.Linq;
+
+namespace DataAccessLayer.Repositories
+{
+    public class SessionRepository
+    {
+        //returns null if no valid session is found in the sessions table, otherwise
+        //  returns the current session
+        public Session GetSession(DatabaseContext _db, string token, Guid userId)

```

Comment:
When getSession() is called the userID will not be accessible.

Reply:
also is token referring to ID of session? If so it is a Universally Unique Identifier so no need to use the UserId as well

Reply:
the token is the actual token the user was given, not the Guid of the token in the session database